showcase.py

```diff
@@ -0,0 +1,81 @@
+from pynamodb.attributes import (
+    UnicodeAttribute, UnicodeSetAttribute, Attribute, LegacyBooleanAttribute,
+    BooleanAttribute, NumberAttribute, BinaryAttribute,MapAttribute, ListAttribute, EnumAttribute)
+
+from pynamodb.models import Model
+
+class MyModel(Model):
+    class Meta:
+        table_name = "TestModel"
+        host="http://localhost:8000"
+    id = UnicodeAttribute(hash_key=True)
+    enum_attr = EnumAttribute(type_set={UnicodeAttribute(null=True), NumberAttribute(null=True), UnicodeSetAttribute(null=True)}, null=True)
+
+if MyModel.exists():
+    MyModel.delete_table()
+MyModel.create_table(read_capacity_units=2, write_capacity_units=2)
+
+
+############# Basic Usage ###############
+# Create an object
+m1 = MyModel("id1")
+m1.enum_attr['UnicodeAttribute'] = "Obama"
+m1.save()
+print (MyModel.dumps())
+# Outputs:
+# [["id1", {"attributes": {"enum_attr": {"S": "{\"S\": \"Obama\"}"}}}]]
+########################################
+
+# Read an object
+m1 = MyModel.get("id1")
+print (m1.enum_attr.get_value())
+# Outputs:
+# Obama
+########################################
+
+# Change value
+m1 = MyModel.get("id1")
+m1.enum_attr["UnicodeAttribute"] = "Obama"
+print (m1.enum_attr["UnicodeAttribute"])
+# Outputs:
+# Obama
+m1.enum_attr["NumberAttribute"] = 2018
+print (m1.enum_attr["NumberAttribute"])
+# Outputs:
+# 2018
+print (m1.enum_attr["UnicodeAttribute"])
+# Outputs:
+# None
+m1.save()
+print (MyModel.dumps())
+# Outputs:
+# [["id1", {"attributes": {"enum_attr": {"S": "{\"N\": \"2018\"}"}}}]]
+########################################
+
+# Different assignment patterns
+m2 = MyModel("id2")
+# All have the same effect
+m2.enum_attr["UnicodeAttribute"] = "President"
```

Comment:
Is this multiple strings way to access things common?  "UnicodeAttribute", "S", "String"

Reply:
the multi-enum-value map theme is new. Usually you access a map like this `map_attr["attr_name"]` or `map_attr.attr_name`. Since I was careful in the design not to add arbitrary names for different values an enum can have, thus we need to access it with a value identifier. Choosing one identifier rather than the other seemed arbitrary to me; that's why I went with these different accessing scheme

Reply:
I'd argue for being opinionated in API design.  Leaving it open is going to leave you with multiple different ways of doing it and having to support it.  I think you should probably pick just one and make it the way.

Reply:
I need to mention though that "S", "String"  are in Pynamo (I am not introducing these)
They are used internally to map the type of the value, (as well as in boto). I have just exposed these identifiers